middleware.js

```diff
@@ -68,7 +68,14 @@ function createEventStream(heartbeat) {
   };
 }
 
-function buildModuleMap(modules) {
+function buildModuleMap(stats) {
+  var modules = stats.modules;
+  if (!modules) {
+    modules = [];
+    stats.children.forEach(function(child) {
```

Comment:
Could you make an `extractModules(stats.children)` function here? I found this a tad tricky to follow at a glance.

Will the modules have unique ids across bundles?